upgpatch: v2ray

Xunop opened the pull request:
Resolved `go test` error; however, due to the current incompatibility of v2ray-core 5.7.0 with go version 1.21, the compilation is still failing.

This patch has been reported to the upstream as issue [v2ray/v2ray-core#2720](https://github.com/v2fly/v2ray-core/issues/2720)

Xeonacid commented:
> Resolved `go test` error; however, due to the current incompatibility of v2ray-core 5.7.0 with go version 1.21, the compilation is still failing.

Thanks for the work! But do you mean it still does not compile with this updated patch? If so, I would suggest you try to fix it together in this PR. After all, a patch that doesn't work is not much meaningful. :P

> This patch has been reported to the upstream as issue [v2ray/v2ray-core#2720](https://github.com/v2fly/v2ray-core/issues/2720)

I'm wondering why do you create an issue with a patch instead of a PR. 🤔

Xunop commented:
> > Resolved `go test` error; however, due to the current incompatibility of v2ray-core 5.7.0 with go version 1.21, the compilation is still failing.
> 
> Thanks for the work! But do you mean it still does not compile with this updated patch? If so, I would suggest you try to fix it together in this PR. After all, a patch that doesn't work is not much meaningful. :P
> 
> > This patch has been reported to the upstream as issue [v2ray/v2ray-core#2720](https://github.com/v2fly/v2ray-core/issues/2720)
> 
> I'm wondering why do you create an issue with a patch instead of a PR. 🤔

emm, I've only fixed the problem with the code itself so far, the compilation error is because the upstream hasn't been updated to v5.8.0 .

You can check the logs [here](https://archriscv.felixc.at/.status/log.htm?url=logs/v2ray/v2ray-5.7.0-1.log), the error I fixed is the one in the logs. But since there is no update upstream (I updated locally to v5.8.0 for testing), so it compile fail. I started out just trying to fix the bug. But I realized that because of the incompatibility of the dependencies downloaded at compile time due to an upstream version change. Fortunately, the upstream has resolved the issue, I can I can fix the bug locally.

Sorry, my expressive ability is not particularly good, so if you have any questions please contact me right away.

Xeonacid commented:
> emm, I've only fixed the problem with the code itself so far, the compilation error is because the upstream hasn't been updated to v5.8.0 .

Then from archriscv side, just waiting for Arch to upgrade should be OK. Also we can make an Arch bug report that it's FTBFS with the new golang.

> You can check the logs [here](https://archriscv.felixc.at/.status/log.htm?url=logs/v2ray/v2ray-5.7.0-1.log), the error I fixed is the one in the logs. But since there is no update upstream (I updated locally to v5.8.0 for testing), so it compile fail. I started out just trying to fix the bug. But I realized that because of the incompatibility of the dependencies downloaded at compile time due to an upstream version change. Fortunately, the upstream has resolved the issue, I can I can fix the bug locally.

TBH I don't see any compile error from the log. Do you mean the failing check?

> Sorry, my expressive ability is not particularly good, so if you have any questions please contact me right away.

No worries at all! Actually you expressed clearly!

Xunop commented:
> > emm, I've only fixed the problem with the code itself so far, the compilation error is because the upstream hasn't been updated to v5.8.0 .
> 
> Then from archriscv side, just waiting for Arch to upgrade should be OK. Also we can make an Arch bug report that it's FTBFS with the new golang.
> 
> > You can check the logs [here](https://archriscv.felixc.at/.status/log.htm?url=logs/v2ray/v2ray-5.7.0-1.log), the error I fixed is the one in the logs. But since there is no update upstream (I updated locally to v5.8.0 for testing), so it compile fail. I started out just trying to fix the bug. But I realized that because of the incompatibility of the dependencies downloaded at compile time due to an upstream version change. Fortunately, the upstream has resolved the issue, I can I can fix the bug locally.
> 
> TBH I don't see any compile error from the log. Do you mean the failing check?

Yep, is the failing check. I just fix those errors. Since go was recently updated to 1.21, v2ay can't be compiled, although v2ray developers have solved this issue, but it's still in pre-realese. if the compilation succeeds and my patch is added, the check fail in the log I mentioned above will be solved. Actually, my patch only solves the check fail problem. So we have to wait for the upstream update, right?
> 
> > Sorry, my expressive ability is not particularly good, so if you have any questions please contact me right away.
> 
> No worries at all! Actually you expressed clearly!

Xeonacid commented:
> So we have to wait for the upstream update, right?

Yeah. And you can also create a bug report to Arch.

Xeonacid commented:
> Thank you for your patience. Do I need to close this pr?

No problem. You may leave it open and we can merge it after Arch upgrades.
